metis-common/src/main/java/eu/europeana/metis/mongo/RecordRedirectDao.java

```diff
@@ -21,28 +21,40 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(RecordRedirectDao.class);
   private static final String OLD_ID = "oldId";
   private static final String NEW_ID = "newId";
+  private Datastore datastore;
 
-  protected MongoClient mongoClient;
-  protected Datastore datastore;
+  /**
+   * Constructor to initialize the mongo mappings/collections and the {@link Datastore} connection.
+   * This constructor is meant to be used when the database is already available.
+   *
+   * @param mongoClient the mongo client connection
+   * @param databaseName the database name of the record redirect database
+   */
+  public RecordRedirectDao(MongoClient mongoClient, String databaseName) {
+    this(mongoClient, databaseName, false);
+  }
 
   /**
-   * Constructs the DAO
-   * <p>Initializes the datastore using the provided mongo client</p>
+   * Constructor to initialize the mongo mappings/collections and the {@link Datastore} connection.
+   * This constructor is meant to be used mostly for when the creation of the database is required.
    *
    * @param mongoClient the mongo client connection
    * @param databaseName the database name of the record redirect database
-   * @param createIndexes flag that enables/disables the creation of the indexes in the database
+   * @param createIndexes flag that initiates the database/indices
    */
   public RecordRedirectDao(MongoClient mongoClient, String databaseName, boolean createIndexes) {
-    this.mongoClient = mongoClient;
+    createDatastore(mongoClient, databaseName);
+    if (createIndexes) {
+      LOGGER.info("Initializing database indices");
+      datastore.ensureIndexes();
+    }
+  }
 
+  private void createDatastore(MongoClient mongoClient, String databaseName) {
```

Comment:
Is it necessary to have this in a separate method? You can integrate this into the constructor and then make the class variable datastore final (or make this method static).
You can do the same in the MorhpiaDatastoreProviderImpl.

Reply:
I though in this way of encapsulating that part, makes the constructor more clear.
Do you feel it's not clear or what you are suggesting is better? I would then change it..

Reply:
It's not important. I think a good way forward would be to make the private method static and return the datastore, so that it can be assigned by the constructor to a final class variable. But we can do that some other time.